presto-main/src/main/java/com/facebook/presto/sql/gen/PageProcessorCompiler.java

```diff
@@ -99,15 +104,18 @@ public PageProcessorCompiler(Metadata metadata)
     @Override
     public void generateMethods(ClassDefinition classDefinition, CallSiteBinder callSiteBinder, RowExpression filter, List<RowExpression> projections)
     {
+        boolean hasFilter = !filter.equals(TRUE_PREDICATE);
```

Comment:
`GenericPageProcessor`doesn't have all the optimizations from `PageProcessorCompiler` (e.g. does not check if projection is identity of constant. Look at `GenericPageProcessor#projectColumnar`